devito/types/dimension.py

```diff
@@ -922,7 +922,7 @@ def _arg_defaults(self, _min=None, size=None, alias=None):
         except AttributeError:
             factor = dim._factor
 
-        defaults[dim.parent.max_name] = range(1, factor*size - 1)
+        defaults[dim.parent.max_name] = range(0, factor*size - 1)
```

Comment:
nitpicking: you can omit the `0, ` part

why was there a 1 before?
is it worth adding a test?

Reply:
I don't think the 1 was there for any specific reason, just a "run at least one step" but not really needed